sqlmesh/core/engine_adapter/base.py

```diff
@@ -612,7 +631,19 @@ def _insert_overwrite_by_condition(
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
     ) -> None:
         table = exp.to_table(table_name)
-        if self.SUPPORTS_INSERT_OVERWRITE:
+        if (
+            self.INSERT_OVERWRITE_STRATEGY.is_delete_insert
```

Comment:
[Nit] `self.INSERT_OVERWRITE_STRATEGY in (InsertOverwriteStrategy.DELETE_INSERT, InsertOverwriteStrategy.REPLACE_WHERE)` maybe